### ISSUE TITLE
Additional Micro Systems and Fixes

### ISSUE THREAD

**Issue by southoz** [aldostools/webMAN-MOD#610]
Hi @aldostools working through the Micros

**Folders in Wiki Missing from code with supporting cores in 1.9.14**

/ROMS/X128 - Commodore C128 - vice_x128_libretro_ps3.SELF
/ROMS/XVIC - Commodore VIC-20 - vice_xvic_libretro_ps3.SELF
/ROMS/XPET - Commodore PET - vice_xpet_libretro_ps3.SELF
/ROMS/XPLUS4 - Commodore PLUS/4 - vice_xplus4_libretro_ps3.SELF
/ROMS/X64SC - Commodore C64 Accurate - vice_x64sc_libretro_ps3.SELF
/ROMS/XCMB2 - Commodore CBM-II - vice_xcmb2_libretro_ps3.SELF

Might be worth adding /ROMS/X64 as VICE points at vice_x64_libretro_ps3.SELF which only supports C64

**Missing extension**

DOSBOX .conf

**Additional Folders supporting Micro Cores in 1.9.14**

/ROMS/CD32 - Amiga CD32 - puae_libretro_ps3.SELF
/ROMS/X64SCPU - Commodore C64 SuperCPU - vice_xscpu64_libretro_ps3.SELF
/ROMS/XCMB25X0 - Commodore CBM-II 5x0 - vice_xcbm5x0_libretro_ps3.SELF

**English Description Updates**

Coleco -> Coleco ColecoVision
Commodore 64, 128 & VIC-20 -> Commodore C64
MSX -> Microsoft MSX
MSX2 -> Microsoft MSX2
Sinclair ZX81 (FUSE) -> Sinclair Spectrum (FUSE) *Note fuse_libretro_ps3.SELF does not support ZX81

The order of the list shown in XMB is not right with the new cores at the end as per the Language file.

**Observation** 

General - Cannot write control remap files and input_remapping_directory = "" changed to /dev_bvd/PS3_GAME/USRDIR/remap works
General - config folder is default with is /dev_hdd/game/PKGLAUNCH/USRDIR
General - Aspect Radio is set to 4:3 instead of core provided.
@crystalct
HATARI - AtariST Disk Folder is mapped to within the Core Configuration (Not Retroarch) by default to /dev_hdd/game/SNE10000/USRDIR/core/system when launching the rom it is not mounted on A: or B: - looks to be hardcoded and can not be changed with the triangle core menu.
HATARI - Has some screen issues if I find a fix other than changing to integer scaling I will let you know

**Comment by aldostools** [comment]
Thank you for your feedback. I'll try to work on these subjects in the next days.

**Comment by aldostools** [comment]
@southoz Most of the suggested changes were implemented in this test build:
Download: [webMAN_MOD_latest.zip](https://github.com/aldostools/webMAN-MOD/files/7962820/webMAN_MOD_latest.zip)

**Comment by southoz** [comment]
@aldostools 

Thanks for the beta,  testing is looking good so far:

Amiga CD32
Commodore C128
Commodore Plus4

Few points:

- MSX description is displaying as MSX/SVI/ColecoVision/SG-1000 (blueMSX) not Microsoft MSX - I had this problem when I manually updated the file in WM_LANG

@crystalct
-Fuse is black screening  there looks to be an issue with the updated core
```
[INFO] Attempt to load executable: [/dev_hdd0/game/RETROARCH/USRDIR/cores/fuse_libretro_ps3.SELF].
[WARN] SELF file is not of NPDRM type, trying another approach to boot it…
```
-Eightone is missing the info file and is not associated with any file types.

**Comment by aldostools** [comment]
@southoz Thank you for your feedback. The issue with MSX looks like the language reader is confusing BMSX with MSX because both contain "MSX" in the name. I moved BMSX down so MSX has a major priority.

You can use the online updater found on the XMB menu or re-download the installer from the release page.

In regards to 81_libretro.info the file is already included in RetroArch CE and it associates the file extensions p|tzx|t81.

**Comment by crystalct** [comment]
> @aldostools
> 
> Thanks for the beta, testing is looking good so far:
> 
> Amiga CD32 Commodore C128 Commodore Plus4
> 
> Few points:
> 
> * MSX description is displaying as MSX/SVI/ColecoVision/SG-1000 (blueMSX) not Microsoft MSX - I had this problem when I manually updated the file in WM_LANG
> 
> @crystalct -Fuse is black screening there looks to be an issue with the updated core
> 
> ```
> [INFO] Attempt to load executable: [/dev_hdd0/game/RETROARCH/USRDIR/cores/fuse_libretro_ps3.SELF].
> [WARN] SELF file is not of NPDRM type, trying another approach to boot it…
> ```
> 
> -Eightone is missing the info file and is not associated with any file types.

Selfs that i make are made using standard `make_self_wc.exe` .... it's just a warning... maybe @aldostools know how avoid it.

**Comment by aldostools** [comment]
Yes, that is only a warning. The cores are standard self.

I copied the method for launch self from Iris Manager.
sysProcessExitSpawn2((const char*)path, (char const**)launchargv, NULL, NULL, 0, 3071, SYS_PROCESS_SPAWN_STACK_SIZE_1M);

**Comment by southoz** [comment]
@crystalct @aldostools it may have something to do with the PKGLauncher using 1.9.0 while the CE package is 1.9.14, I will do some more testing and see what I can turn up.

**Comment by aldostools** [comment]
1.9.0 and CE install in different directories.

PKGLAUNCH searches first CE and only tries to use RetroArch 1.9.0 if RetroArch CE is not installed.

**Comment by southoz** [comment]
I will definitely test this again but my experience on defaulted PlayStation and the latest CE is

- CE BIN is 1.9.14
- CE Can not find assets when started through XMB
- PKGLauncher BIN is 1.9.0
- PKGLauncher cannot save remaps un directory is updated

Order of installation was WebMAN-MOD with an update, Retroarch CE and PKGLauncher

**Comment by aldostools** [comment]
RetroArch CE has mixed versions (some cores are 1.9.14 others are 1.9.0)

When detects RetroArch CE installed, the default retroarch.cfg in /dev_hdd0/game/PKGLAUNCH/PS3_GAME/USRDIR is replaced with [retroarch.cce](https://github.com/aldostools/webMAN-MOD/blob/master/_Projects_/wm_url_launcher/pkg/PS3_GAME/USRDIR/retroarch.cce)

In this file assets_directory = "/dev_hdd0/game/RETROARCH/USRDIR/assets"

When the assets are not loaded, the XMB icons are not displayed in RetroArch:
![image](https://user-images.githubusercontent.com/974897/151898689-9665beed-04b8-464b-8897-0ce70331e33f.png)

**Comment by crystalct** [comment]
Let me update all working cores to 1.10.x and then we will have the next version of Retroarch CE.....
i'm fixing a couple of things

**Comment by southoz** [comment]
No rush, do you have a Sharp x68000

**Comment by southoz** [comment]
@aldostools due to the release of beta 4 I am going to have to restart the testing and realign the cores.

So far the following cores are included that were not in 1.9.14
Bettle SuperGrafx
Genesis Plus GX Wide
Picodrive (Should add 32x Support)
EightOne (Was included but the ELF was not recognised)
VICE DRV
Mame 2003 Midway

**Comment by aldostools** [comment]
FYI I have updated recently PKGLAUNCH to support the new cores in beta 0.4, among other fixes in webMAN MOD.
